bin/ROC_Efficiency.cc

```diff
@@ -0,0 +1,235 @@
+////////////////////////////////////////////////////////////////////
+//
+// Dean Andrew Hidas <Dean.Andrew.Hidas@cern.ch>
+//
+// Created on: Mon Jul  4 19:20:41 CEST 2011
+//
+////////////////////////////////////////////////////////////////////
+//Edited by Thoth Gunter
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <cmath>
+
+#include "PLTEvent.h"
+#include "PLTU.h"
+
+
+
+class HitCounter
+{
+  public:
+    HitCounter () {
+      NFiducial[0] = 0;
+      NFiducial[1] = 0;
+      NFiducial[2] = 0;
+      NFiducialAndHit[0] = 0;
+      NFiducialAndHit[1] = 0;
+      NFiducialAndHit[2] = 0;
+    }
+    ~HitCounter () {}
+
+    int NFiducial[3];
+    int NFiducialAndHit[3];
+};
+
+
+
+// FUNCTION DEFINITIONS HERE
+int TrackingEfficiency (std::string const, std::string const, std::string const);
+
+
+
+
+
+
+
+// CODE BELOW
+
+int TrackingEfficiency (std::string const DataFileName, std::string const GainCalFileName, std::string const AlignmentFileName)
+{
+  std::cout << "DataFileName:      " << DataFileName << std::endl;
+  std::cout << "AlignmentFileName: " << AlignmentFileName << std::endl;
+
+  // Set some basic style for plots
+  PLTU::SetStyle();
+
+  // Grab the plt event reader
+  PLTEvent Event(DataFileName, GainCalFileName, AlignmentFileName);
+
+  PLTPlane::FiducialRegion FidRegionHits  = PLTPlane::kFiducialRegion_Diamond;
+  PLTPlane::FiducialRegion FidRegionTrack = PLTPlane::kFiducialRegion_m5_m5;//<-Original
```

Comment:
This should really be changed. First, although kFiducialRegion_Diamond is equivalent to kFiducialRegion_All these days, I would still use the latter to avoid confusion. Second, kFiducualRegion_m5_m5 just drops the outermost five rows and columns on the sensor. For most of the ROCs, this will have no effect at all since these are already masked. But if there is a ROC where the active area is in this region (I don't know offhand if there is), then this will cause different behavior for this ROC, which is probably undesirable. I would recommend just making these both kFiducialRegion_All.

If you want to apply the mask, the correct way to do it is to take the online mask (Mask_2016_VdM_v1.txt) and apply it using PLTBinaryFileReader::ReadOnlinePixelMask(). But as a practical matter, there's not really much point to applying the mask offline -- if the pixel data is decoded correctly, applying the mask won't do anything, and if it's not, then you have much bigger problems anyway.